### ISSUE TITLE
Unable to install sumneko_lua

### ISSUE THREAD

**Issue by AvlKP-dev** [williamboman/nvim-lsp-installer#271]
**Describe the bug**
:LspInstall sumneko_lua is throwing an error.

**Expected behavior**
:LspInstall successfully installed sumneko_lua.

**Screenshots**
![WindowsTerminal_l6jtdu9HG3](https://user-images.githubusercontent.com/79994953/142625109-a1471340-18fa-44c7-b2a3-2e7e1873c084.png)

**System info (please complete the following information):**
 - OS: Windows 10
 - Neovim version: 0.5.1

Please also include any other system information you think is relevant!

**Install log**
If applicable, include nvim-lsp-installer's logs. See `:help nvim-lsp-installer-debugging` for instructions on how to set the log level and access the logs.

```
[DEBUG 11/19/2021 7:44:13 PM] ...-installer\lua\nvim-lsp-installer\ui\status-win\init.lua:616: Queuing server="sumneko_lua", version="" for installation
[DEBUG 11/19/2021 7:44:13 PM] ...nvim-lsp-installer\lua\nvim-lsp-installer\ui\display.lua:303: Buffer or window is no longer valid 1004 4
[INFO  11/19/2021 7:44:13 PM] ...-installer\lua\nvim-lsp-installer\ui\status-win\init.lua:551: Starting install server_name="sumneko_lua", requested_version=""
[DEBUG 11/19/2021 7:44:13 PM] ...r\start\nvim-lsp-installer\lua\nvim-lsp-installer\fs.lua:41: fs: mkdirp C:\Users\Avila\AppData\Local\nvim-data\lsp_servers\sumneko_lua.tmp
[DEBUG 11/19/2021 7:44:13 PM] ...-installer\lua\nvim-lsp-installer\installers\context.lua:13: Fetching URL "https://api.github.com/repos/sumneko/vscode-lua/releases/latest"
[DEBUG 11/19/2021 7:44:13 PM] ...rt\nvim-lsp-installer\lua\nvim-lsp-installer\process.lua:116: Spawning cmd="powershell.exe", spawn_opts={
  args = { "-NoProfile", "-Command", "$ProgressPreference = 'SilentlyContinue';Write-Output (iwr -UseBasicParsing -Uri \"https://api.github.com/repos/sumneko/vscode-lua/releases/latest\").Content" }
}
[DEBUG 11/19/2021 7:44:13 PM] ...rt\nvim-lsp-installer\lua\nvim-lsp-installer\process.lua:162: Spawned with pid 14028
[DEBUG 11/19/2021 7:44:13 PM] ...nvim-lsp-installer\lua\nvim-lsp-installer\ui\display.lua:396: Opening window
[DEBUG 11/19/2021 7:44:13 PM] ...rt\nvim-lsp-installer\lua\nvim-lsp-installer\process.lua:148: Job pid=14028 exited with exit_code=0, signal=0
[DEBUG 11/19/2021 7:44:13 PM] ...-installer\lua\nvim-lsp-installer\installers\context.lua:16: Successfully fetched URL "https://api.github.com/repos/sumneko/vscode-lua/releases/latest"
[DEBUG 11/19/2021 7:44:13 PM] ...-installer\lua\nvim-lsp-installer\installers\context.lua:83: Resolved latest version sumneko_lua sumneko/vscode-lua v2.4.9
[DEBUG 11/19/2021 7:44:13 PM] ...rt\nvim-lsp-installer\lua\nvim-lsp-installer\process.lua:116: Spawning cmd="powershell.exe", spawn_opts={
  args = { "-NoProfile" },
  cwd = "C:\\Users\\Avila\\AppData\\Local\\nvim-data\\lsp_servers\\sumneko_lua.tmp",
  env = { "PROCESSOR_REVISION=<redacted>", "VIMRUNTIME=<redacted>", "CHOCOLATEYLASTPATHUPDATE=<redacted>", "CONDA_PROMPT_MODIFIER=<redacted>", "PROCESSOR_LEVEL=<redacted>", "MYVIMRC=<redacted>", "SYSTEMROOT=<redacted>", "COMPUTERNAME=<redacted>", "DRIVERDATA=<redacted>", "PROGRAMW6432=<redacted>", "PATHEXT=<redacted>", "WINDIR=<redacted>", "WSLENV=<redacted>", "OS=<redacted>", "HOMEDRIVE=<redacted>", "ONEDRIVE=<redacted>", "COMMONPROGRAMFILES=<redacted>", "WT_SESSION=<redacted>", "WT_PROFILE_ID=<redacted>", "USERPROFILE=<redacted>", "PROGRAMFILES(X86)=<redacted>", "NVIM_LISTEN_ADDRESS=<redacted>", "PROGRAMDATA=<redacted>", "APPDATA=<redacted>", "POWERSHELL_DISTRIBUTION_CHANNEL=<redacted>", "VIM=<redacted>", "PROCESSOR_ARCHITECTURE=<redacted>", "CHOCOLATEYTOOLSLOCATION=<redacted>", "JD2_HOME=<redacted>", "HOMEPATH=<redacted>", "COMMONPROGRAMFILES(X86)=<redacted>", "SYSTEMDRIVE=<redacted>", "TEMP=<redacted>", "LOGONSERVER=<redacted>", "ZES_ENABLE_SYSMAN=<redacted>", "JAVA_HOME=<redacted>", "PROCESSOR_IDENTIFIER=<redacted>", "USERNAME=<redacted>", "TMP=<redacted>", "PSMODULEPATH=<redacted>", "NVIM_LOG_FILE=<redacted>", "USERDOMAIN=<redacted>", "AZ_ENABLED=<redacted>", "PUBLIC=<redacted>", "ALLUSERSPROFILE=<redacted>", "COMSPEC=<redacted>", "GOPATH=C:\\Users\\Avila\\go", "USERDOMAIN_ROAMINGPROFILE=<redacted>", "COMMONPROGRAMW6432=<redacted>", "PROGRAMFILES=<redacted>", "LOCALAPPDATA=<redacted>", "TERM=<redacted>", "POSH_GIT_ENABLED=<redacted>", "POWERLINE_COMMAND=<redacted>", "NUMBER_OF_PROCESSORS=<redacted>", "PATH=C:\\Users\\Avila\\scoop\\apps\\gcc\\current\\bin;C:\\Program Files\\PowerShell\\7;C:\\Program Files\\Volta\\;C:\\WINDOWS\\system32;C:\\WINDOWS;C:\\WINDOWS\\System32\\Wbem;C:\\WINDOWS\\System32\\WindowsPowerShell\\v1.0\\;C:\\WINDOWS\\System32\\OpenSSH\\;C:\\Program Files\\Git\\cmd;C:\\Users\\Avila\\AppData\\Local\\Programs\\Python\\Python39;C:\\Users\\Avila\\AppData\\Local\\Programs\\Python\\Python39\\Scripts;C:\\Program Files\\dotnet\\;C:\\Program Files\\jdk-17\\bin;C:\\Program Files\\PowerShell\\7\\;C:\\Program Files\\Go\\bin;C:\\Users\\Avila\\lua-language-server\\3rd\\luamake;C:\\Users\\Avila\\scoop\\apps\\oh-my-posh\\current\\bin;C:\\Users\\Avila\\scoop\\shims;C:\\Users\\Avila\\AppData\\Local\\Volta\\bin;C:\\Users\\Avila\\AppData\\Local\\Microsoft\\WindowsApps;E:\\MiKTeX\\texmfs\\install\\miktex\\bin\\x64;E:\\pdf2svg;C:\\Users\\Avila\\AppData\\Local\\Pandoc\\;C:\\Users\\Avila\\go\\bin;C:\\Users\\Avila\\scoop\\apps\\neovim\\current\\bin" }
}
[DEBUG 11/19/2021 7:44:13 PM] ...rt\nvim-lsp-installer\lua\nvim-lsp-installer\process.lua:162: Spawned with pid 8380
[DEBUG 11/19/2021 7:44:16 PM] ...rt\nvim-lsp-installer\lua\nvim-lsp-installer\process.lua:148: Job pid=8380 exited with exit_code=0, signal=0
[DEBUG 11/19/2021 7:44:16 PM] ...rt\nvim-lsp-installer\lua\nvim-lsp-installer\process.lua:116: Spawning cmd="powershell.exe", spawn_opts={
  args = { "-NoProfile" },
  cwd = "C:\\Users\\Avila\\AppData\\Local\\nvim-data\\lsp_servers\\sumneko_lua.tmp",
  env = { "PROCESSOR_REVISION=<redacted>", "VIMRUNTIME=<redacted>", "CHOCOLATEYLASTPATHUPDATE=<redacted>", "CONDA_PROMPT_MODIFIER=<redacted>", "PROCESSOR_LEVEL=<redacted>", "MYVIMRC=<redacted>", "SYSTEMROOT=<redacted>", "COMPUTERNAME=<redacted>", "DRIVERDATA=<redacted>", "PROGRAMW6432=<redacted>", "PATHEXT=<redacted>", "WINDIR=<redacted>", "WSLENV=<redacted>", "OS=<redacted>", "HOMEDRIVE=<redacted>", "ONEDRIVE=<redacted>", "COMMONPROGRAMFILES=<redacted>", "WT_SESSION=<redacted>", "WT_PROFILE_ID=<redacted>", "USERPROFILE=<redacted>", "PROGRAMFILES(X86)=<redacted>", "NVIM_LISTEN_ADDRESS=<redacted>", "PROGRAMDATA=<redacted>", "APPDATA=<redacted>", "POWERSHELL_DISTRIBUTION_CHANNEL=<redacted>", "VIM=<redacted>", "PROCESSOR_ARCHITECTURE=<redacted>", "CHOCOLATEYTOOLSLOCATION=<redacted>", "JD2_HOME=<redacted>", "HOMEPATH=<redacted>", "COMMONPROGRAMFILES(X86)=<redacted>", "SYSTEMDRIVE=<redacted>", "TEMP=<redacted>", "LOGONSERVER=<redacted>", "ZES_ENABLE_SYSMAN=<redacted>", "JAVA_HOME=<redacted>", "PROCESSOR_IDENTIFIER=<redacted>", "USERNAME=<redacted>", "TMP=<redacted>", "PSMODULEPATH=<redacted>", "NVIM_LOG_FILE=<redacted>", "USERDOMAIN=<redacted>", "AZ_ENABLED=<redacted>", "PUBLIC=<redacted>", "ALLUSERSPROFILE=<redacted>", "COMSPEC=<redacted>", "GOPATH=C:\\Users\\Avila\\go", "USERDOMAIN_ROAMINGPROFILE=<redacted>", "COMMONPROGRAMW6432=<redacted>", "PROGRAMFILES=<redacted>", "LOCALAPPDATA=<redacted>", "TERM=<redacted>", "POSH_GIT_ENABLED=<redacted>", "POWERLINE_COMMAND=<redacted>", "NUMBER_OF_PROCESSORS=<redacted>", "PATH=C:\\Users\\Avila\\scoop\\apps\\gcc\\current\\bin;C:\\Program Files\\PowerShell\\7;C:\\Program Files\\Volta\\;C:\\WINDOWS\\system32;C:\\WINDOWS;C:\\WINDOWS\\System32\\Wbem;C:\\WINDOWS\\System32\\WindowsPowerShell\\v1.0\\;C:\\WINDOWS\\System32\\OpenSSH\\;C:\\Program Files\\Git\\cmd;C:\\Users\\Avila\\AppData\\Local\\Programs\\Python\\Python39;C:\\Users\\Avila\\AppData\\Local\\Programs\\Python\\Python39\\Scripts;C:\\Program Files\\dotnet\\;C:\\Program Files\\jdk-17\\bin;C:\\Program Files\\PowerShell\\7\\;C:\\Program Files\\Go\\bin;C:\\Users\\Avila\\lua-language-server\\3rd\\luamake;C:\\Users\\Avila\\scoop\\apps\\oh-my-posh\\current\\bin;C:\\Users\\Avila\\scoop\\shims;C:\\Users\\Avila\\AppData\\Local\\Volta\\bin;C:\\Users\\Avila\\AppData\\Local\\Microsoft\\WindowsApps;E:\\MiKTeX\\texmfs\\install\\miktex\\bin\\x64;E:\\pdf2svg;C:\\Users\\Avila\\AppData\\Local\\Pandoc\\;C:\\Users\\Avila\\go\\bin;C:\\Users\\Avila\\scoop\\apps\\neovim\\current\\bin" }
}
[DEBUG 11/19/2021 7:44:16 PM] ...rt\nvim-lsp-installer\lua\nvim-lsp-installer\process.lua:162: Spawned with pid 14172
[DEBUG 11/19/2021 7:44:17 PM] ...rt\nvim-lsp-installer\lua\nvim-lsp-installer\process.lua:148: Job pid=14172 exited with exit_code=1, signal=0
[DEBUG 11/19/2021 7:44:17 PM] ...r\start\nvim-lsp-installer\lua\nvim-lsp-installer\fs.lua:22: fs: rmrf C:\Users\Avila\AppData\Local\nvim-data\lsp_servers\sumneko_lua.tmp
[INFO  11/19/2021 7:44:17 PM] ...-installer\lua\nvim-lsp-installer\ui\status-win\init.lua:570: Installation completed server_name="sumneko_lua", success=false
```

**Comment by TeoDev1611** [comment]
@AvlKP-dev  What's your version of powershell and are you running on x86 or x64 machine? 

```ps1
Install-Module PSReadLine
```

You can run this in Powershell and this fix the bug i think :)

**Comment by AvlKP-dev** [comment]
I configured :term to use PowerShell Core 7.2.0 following to :help shell-powershell settings. The machine is x64. I have tried installing the module on both the PowerShell Core and Windows PowerShell. Both of it already had the module installed so I reinstalled it. It still doesn't work.

**Comment by d9d6ka** [comment]
I've managed to install sumneko_lua. I've peeked into `lua\nvim-lsp-installer\installers\shell.lua` and replaced
```lua
shell = "powershell.exe",
```
with
```lua
shell = "pwsh",
```
I have PowerShell Core 7.2.0 installed, btw.

![2021-11-20_21-07-35](https://user-images.githubusercontent.com/29946047/142736731-6a48f087-3226-448e-8eb0-5b97c938fa82.png)

*Suggestion*: maybe it has sense to detect whether PowerShell Core installed.

**Comment by williamboman** [comment]
Hello! I'll have to rely on Google for this as I don't use Windows. I see that you have a `PSMODULEPATH` environment variable defined - what happens if you remove that variable in a shell session and then start neovim in the same session and attempt to install?

Are you otherwise able to run `Expand-Archive` in a `powershell.exe` session? What happens if you download https://github.com/sumneko/vscode-lua/releases/download/v2.4.9/lua-2.4.9.vsix and manually try to `Expand-Archive` that file?

> _Suggestion_: maybe it has sense to detect whether PowerShell Core installed.

I find the whole powershell vs pwsh really confusing - which one can or should be assumed to exist by default? I've been under the impression that powershell.exe pretty much always ships with Windows, and pwsh is a new version that you manually need to install.  This reminds me of a discussion around issues installing on Windows [here](https://github.com/williamboman/nvim-lsp-installer/issues/201), which we resolved by simply adding the `-NoProfile` flag.

**Comment by d9d6ka** [comment]
BTW, I have Debian on a virtual machine, everything works here like a charm! :)

> Hello! I'll have to rely on Google for this as I don't use Windows. I see that you have a `PSMODULEPATH` environment variable defined - what happens if you remove that variable in a shell session and then start neovim in the same session and attempt to install?

It works. After removing the env variable from the session `sumneko_lua` is uninstalled and installed again in the same session successfully.

> Are you otherwise able to run `Expand-Archive` in a `powershell.exe` session? What happens if you download https://github.com/sumneko/vscode-lua/releases/download/v2.4.9/lua-2.4.9.vsix and manually try to `Expand-Archive` that file?

Old `powershell.exe` included into Windows 10 fails to Expand-Archive as it understands only zip-files.
```powershell
Expand-Archive : .vsix is not a supported archive file format. .zip is the only supported archive file format.
At line:1 char:1
+ Expand-Archive .\lua-2.4.9.vsix
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (.vsix:String) [Expand-Archive], IOException
    + FullyQualifiedErrorId : NotSupportedArchiveFileExtension,Expand-Archive
```

PowerShell Core extracts the archive with no problems.

> I find the whole powershell vs pwsh really confusing - which one can or should be assumed to exist by default? I've been under the impression that powershell.exe pretty much always ships with Windows, and pwsh is a new version that you manually need to install. This reminds me of a discussion around issues installing on Windows [here](https://github.com/williamboman/nvim-lsp-installer/issues/201), which we resolved by simply adding the `-NoProfile` flag.

I understand. Maybe it would be more convenient to add modern PowerShell as a recommendation. The installation already requires some extra programs (e.g. Python, npm...)

**Comment by TeoDev1611** [comment]
> BTW, I have Debian on a virtual machine, everything works here like a charm! :)
> 
> > Hello! I'll have to rely on Google for this as I don't use Windows. I see that you have a `PSMODULEPATH` environment variable defined - what happens if you remove that variable in a shell session and then start neovim in the same session and attempt to install?
> 
> It works. After removing the env variable from the session `sumneko_lua` is successfully uninstalled and installed again in the same session.
> 
> > Are you otherwise able to run `Expand-Archive` in a `powershell.exe` session? What happens if you download https://github.com/sumneko/vscode-lua/releases/download/v2.4.9/lua-2.4.9.vsix and manually try to `Expand-Archive` that file?
> 
> Old `powershell.exe` included into Windows 10 fails to Expand-Archive as it understands only zip-files.
> 
> ```powershell
> Expand-Archive : .vsix is not a supported archive file format. .zip is the only supported archive file format.
> At line:1 char:1
> + Expand-Archive .\lua-2.4.9.vsix
> + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>     + CategoryInfo          : InvalidArgument: (.vsix:String) [Expand-Archive], IOException
>     + FullyQualifiedErrorId : NotSupportedArchiveFileExtension,Expand-Archive
> ```
> 
> PowerShell Core extracts the archive with no problems.
> 
> > I find the whole powershell vs pwsh really confusing - which one can or should be assumed to exist by default? I've been under the impression that powershell.exe pretty much always ships with Windows, and pwsh is a new version that you manually need to install. This reminds me of a discussion around issues installing on Windows [here](https://github.com/williamboman/nvim-lsp-installer/issues/201), which we resolved by simply adding the `-NoProfile` flag.
> 
> I understand. Maybe it would be more convenient to add modern PowerShell as a recommendation. The installation already requires some extra programs (e.g. Python, npm...)

The new powershell dont support the .vsix file can you test this:

```ps1
Rename-Item -Path .\lua-2.4.9.vsix -NewName .\lua-2.4.9.zip
Expand-Archive  .\lua-2.4.9.zip
```

This can fix and extract into a folder for the installer only add the `Rename-Item` command 😃

**Comment by williamboman** [comment]
I'd be happy to attempt `pwsh` first, if such an executable exists. How different are `powershell` and `pwsh` in terms of syntax and available tools? The usage of PS in this plugin is very basic, so I'm not worried it'll cause issues but you never know.

**Comment by williamboman** [comment]
Actually scratch that. I'd like to stick to `powershell` as much as possible - I don't think a `pwsh -> powershell` fallback is warranted given the limited and basic usage of it.

> It works. After removing the env variable from the session `sumneko_lua` is successfully uninstalled and installed again in the same session.

If this works, I suggest we simply exclude the `PSMODULEPATH` environment variable when spawning powershell commands. I'll open a PR

**Comment by williamboman** [comment]
What about branch `fix-271` (https://github.com/williamboman/nvim-lsp-installer/pull/278)?

**Comment by d9d6ka** [comment]
> What about branch `fix-271` (#278)?

It turns out that @AvlKP-dev has forgot to mention his version of PowerShell. **Because it matters.**
I've made a mistake as well as I've messed versions a bit. Shame on me :cry:

I have two versions: built-in 5.1 and newer Core 7.2.0.
If I start nvim from `cmd.exe` or `powershell.exe` session everything works as expected both with and without the patch #278.
If I start nvim from the new PowerShell Core `pwsh.exe` session, then I need to delete `PSMODULEPATH` from the session, and then everything works. Without the deletion I receive the same result as @AvlKP-dev does.
